src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_formula_generator.py

```diff
@@ -140,3 +140,60 @@ def generate(
         self,
     ) -> FormulaEngine[QuantityT] | FormulaEngine3Phase[QuantityT]:
         """Generate a formula engine, based on the component graph."""
+
+    def _get_metric_fallback_components(
+        self, components: set[Component]
+    ) -> dict[Component, set[Component]]:
+        """Get primary and fallback components within a given set of components.
+
+        This function assumes that fallback component has exactly one meter
+        in from of him.
```

Comment:
> has exactly one meter in from of him.

..what?

Reply:
Ooh, one *Meter*, not the distance.  This whole method description was very confusing to me..

Reply:
I actually didn't even notice the missing "front"  XD for me the confusion was more "him.. who is he?". 
Maybe writing "metering component" or so would make it more clear

Reply:
Function description was changed.